prisma/schema.prisma

```diff
@@ -76,39 +76,46 @@ model LaborMarket {
 }
 
 model ServiceRequest {
-  id                 String       @id @default(cuid())
+  internalId         String       @default(cuid())
   title              String
   laborMarketAddress String
   createdAt          DateTime     @default(now())
   laborMarket        LaborMarket  @relation(fields: [laborMarketAddress], references: [address])
   submissions        Submission[]
+
+  @@id([internalId, laborMarketAddress])
 }
 
 model Review {
-  id           String     @id @unique
-  submissionId String
-  comment      String
-  score        Int
-  createdAt    DateTime   @default(now())
-  creatorId    String
-  submission   Submission @relation(fields: [submissionId], references: [id])
+  id                 String      @id @unique
+  submissionId       String
+  comment            String
+  createdAt          DateTime    @default(now())
+  creatorId          String
+  serviceRequestId   String
+  laborMarketAddress String
+  submission         Submission? @relation(fields: [submissionId, laborMarketAddress], references: [internalId, laborMarketAddress])
+  score              Int
 }
 
 // Represents an on-chain submission to a service request.
 model Submission {
-  id               String         @id @unique
-  serviceRequestId String
-  createdAt        DateTime       @default(now())
-  creatorId        String
-  title            String
-  description      String
-  score            Int
-  serviceRequest   ServiceRequest @relation(fields: [serviceRequestId], references: [id])
-  reviews          Review[]
+  internalId         String         @default(cuid())
+  serviceRequestId   String
+  createdAt          DateTime       @default(now())
+  creatorId          String
+  title              String
+  description        String
+  laborMarketAddress String
+  score              Int
+  reviews            Review[]
+  serviceRequest     ServiceRequest @relation(fields: [laborMarketAddress, serviceRequestId], references: [laborMarketAddress, internalId])
+
+  @@id([internalId, laborMarketAddress])
```

Comment:
id needs to include the internalId in case of a db reset. So like `[laborMarketAddress, serviceSubmissionId]`. Also thinking that internalId can probably just be called `id`.

Reply:
weird Prisma issue with the name id because of the composite key also being id, will ping you about it

Reply:
maybe I am wrong actually, let me double check

Reply:
@sebastian will that just be a string? what point will that get added?

Reply:
That'll get added when at the time of indexing